caikit/interfaces/common/data_model/remote.py

```diff
@@ -0,0 +1,139 @@
+# Copyright The Caikit Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This file contains interfaces required to connect to Remote servers
```

Comment:
@gabe-l-hart so that was actually the goal with the `ConnectionInfo` dataobject. I had hoped to reuse it for other remote connections like S3

Reply:
Ah! Makes sense. Also makes sense why you moved `protocol`. One thought would be to subclass a base `Connection` to include `protocol` in a `CaikitRuntimeConnection`?

Reply:
Hmmmmm, I'm not sure if `CaikitRuntimeConnection` is needed for `protocol` because at that point should all of the RemoteModuleFinder's config live in a DataObject?